src/object.d

```diff
@@ -1703,13 +1705,17 @@ class TypeInfo_Struct : TypeInfo
 {
     override string toString() const { return name; }
 
+    override size_t toHash() const
+    {
+        return hashOf(this.mangledName);
+    }
+
```

Comment:
`TypeInfo.opCmp` uses `toString()` too; I'm not convinced overriding it here to check for a `TypeInfo_Struct` rhs and comparing the `mangledName` instead is worth it.